### ISSUE TITLE
ActorSystem terminates when restarted remote ActorSystem has associated after being quarantined

### ISSUE THREAD

**Issue by RickLatrine** [akka/akka#17213]
I have two ActorSystems Node-A and Node-B in separate JVM processes with netty-tcp remoting enabled running on the same machine (127.0.0.1) on port 2552 and 2553.

Each ActorSystem starts itself a actor.
The actor sends a "ping" message and replies to a sender with a "pong" message using actorSelection which identifies the corresponding remote actor.
On receiving a "ping" or "pong" message the sender actor is going to be watched.

Everything works fine so far.

Then I kill the JVM process which was running ActorSystem Node-B.
Node-A detects the termination and after a while the remote system Node-B becomes quarantined.

Now I restart the JVM process to launch the ActorSystem Node-B.
The ActorSystem Node-A notifies that Node-B is associated.
In the same instant ActorSystem Node-A terminates without any log message or exception.
Node-B continues and notifies about that termination

This unexpected behaviour is reproducible.

Further investigation shows up that the method akka.remote.EndpointManager#createEndpoint contains an assertion:
assert(writing || refuseUid.isEmpty)

If an endpoint which was previously quarantined reconnects, refuseUid is existing.
As the remote endpoint sends a message, a reading endpoint must be created, so writing is false.

The resulting AssertionError smashes through all supervising parents until the ActorSystem terminates.

See also: https://groups.google.com/forum/#!topic/akka-user/jleFC7P66ao

**Comment by RickLatrine** [comment]
I re-tested the issue with akka versions 2.3.7 and 2.3.8.
The error is reproducible from version 2.3.8 on.

In version 2.3.7 no QuarantinedEvent will be published.
At the assert statement in method createEndpoint the refuseUid is empty in any case.

So, I will go with version 2.3.7 until this issue is fixed.

**Comment by drewhk** [comment]
Thanks for reporting!

**Comment by drewhk** [comment]
Reproduced with a test. Thanks for the description!

This problem arises less likely if passive connections are turned off, because this assertion only triggers if we get a reading endpoint actor from a host that has been quarantined before. The assertion probably does not make sense at all, but I need to investigate a bit more to see if it is really safe to remove.

**Comment by RickLatrine** [comment]
Yes, git annotations/blame told me that this might be relict due to a refactoring some time ago...

**Comment by drewhk** [comment]
Ok, I tested manually if removing the assertion works - it does. I also tested if the system is not restarted, and it creates a connection it still remains quarantined - it does. So I conclude that the assertion makes no sense at all, the refuseUid field is used correctly even for the read-only endpoint actors. I remove that line, and change all remaining assertions (if there are any) to normal exceptions, so at least things are logged.

I will also add extra cleanup in postStop.
